crates/octez/src/async/client.rs

```diff
@@ -331,6 +332,81 @@ impl OctezClient {
         self.spawn_and_wait_command(args).await?;
         Ok(())
     }
+
+    pub async fn originate_contract(
+        &self,
+        contract_alias: &str,
+        funding_account_src: &str, // can be an address or an alias
+        fund_tez: f64,
+        contract_path: &Path,
+        init_data: Option<&str>,
+        burn_cap: Option<f64>,
+    ) -> Result<(ContractKt1Hash, OperationHash)> {
+        let fund_str = fund_tez.to_string();
+        let burn_cap_str = match burn_cap {
+            Some(v) => v.to_string(),
+            None => String::new(),
+        };
+        let mut args = vec![
+            "originate",
+            "contract",
+            contract_alias,
+            "transferring",
+            &fund_str,
+            "from",
+            funding_account_src,
+            "running",
+            contract_path
+                .to_str()
+                .ok_or(anyhow::anyhow!("failed to convert contract path to string"))?,
+        ];
+        if let Some(data) = init_data {
+            args.append(&mut vec!["--init", data]);
+        }
+        if !burn_cap_str.is_empty() {
+            args.append(&mut vec!["--burn-cap", &burn_cap_str]);
+        }
+        let output = self.spawn_and_wait_command(args).await?;
```

Comment:
`burn_cap` is provided as an Optional arg, there is no need to transform it to a string on line 346 then test its presence with `is_empty`? Please could you use `if let Some` here instead and remove line 346-348.

Reply:
It's like this because rustc complains that the option string being pushed does not live long enough if I use `if let Some`.

Reply:
Ah :(